src/main/java/org/sagebionetworks/bridge/spring/controllers/StudyParticipantController.java

```diff
@@ -142,11 +149,15 @@ final void setSchedule2Service(Schedule2Service scheduleService) {
         if (isUpToDate(modifiedSince, modifiedOn)) {
             return new ResponseEntity<>(NOT_MODIFIED);
         }
-        App app = appService.getApp(session.getAppId());
         Schedule2 schedule = scheduleService.getScheduleForStudy(session.getAppId(), study);
         cacheProvider.setObject(scheduleModificationTimestamp(studyId), schedule.getModifiedOn().toString());
         
-        activityEventService.publishActivitiesRetrieved(app, studyId, session.getHealthCode(), DateTime.now());
+        studyActivityEventService.publishEvent(new StudyActivityEventRequest()
+                .appId(session.getAppId())
+                .studyId(studyId)
+                .userId(session.getId())
+                .objectType(TIMELINE_RETRIEVED)
+                .timestamp(getDateTime()));
```

Comment:
What about first sign in as an event?

Reply:
Consider removing study_start_date since it is in fact mutable.

This scenario is problematic: download events, then download timeline, and the study_start_date will be different because timeline was retrieved (all of a sudden, order of operations matter).

Reply:
Another option: publish this if you retrieve activity events (and include it in that map) and/or when you retrieve the timeline, so they are not out-of-sync.

Reply:
study_start_date removed. timeline_retrieved added to both the study activities and timeline endpoints. first sign in I am not adding because there's nothing about it that's specific to any study. We'd have to record it as a global event, then retrieve it and include it in the study-specific events...enrollment or timeline_retrieved would be better.